src/python/pants/backend/python/tasks/select_interpreter.py

```diff
@@ -65,15 +66,17 @@ def execute(self):
     if not python_tgts_and_reqs:
       return
     python_tgts = [tgt for tgt in python_tgts_and_reqs if isinstance(tgt, PythonTarget)]
-    fs = PythonInterpreterFingerprintStrategy()
+    interpreter_cache = PythonInterpreterCache.global_instance()
+    fs = PythonInterpreterFingerprintStrategy(python_setup=interpreter_cache._python_setup)
     with self.invalidated(python_tgts, fingerprint_strategy=fs) as invalidation_check:
-      # If there are no relevant targets, we still go through the motions of selecting
-      # an interpreter, to prevent downstream tasks from having to check for this special case.
-      if invalidation_check.all_vts:
-        target_set_id = VersionedTargetSet.from_versioned_targets(
-            invalidation_check.all_vts).cache_key.hash
-      else:
-        target_set_id = 'no_targets'
+      # If there are no constraints, meaning no global constraints nor compatibility requirements on
+      # the targets, we still go through the motions of selecting an interpreter, to prevent
+      # downstream tasks from having to check for this special case.
+      target_set_id = (
+        'no_constraints'
```

Comment:
I renamed this because semantically it is no longer `no_targets`, which meant no targets with `compatibility`. Instead, this case only triggers if there are both no global constraints and no `compatibility` entries for the targets.

Do I need to bump the `implementation_version` as a result?

Reply:
I believe you would need to bump the `implementation_version` anyway since we are fixing the larger caching error in this task, but I think anything that modifies what files are in the cache dir merits bumping it (so yes, I think).